### ISSUE TITLE
Plugin Update mechanism seems to be broken

### ISSUE THREAD

**Issue by blackcoffeerider** [ninja-ide/ninja-ide-web#24]
Problem: When a plugin-archive is uploaded under the same name as an already existing one and the version is incremented - the more recent file gets created as "seperate instance" of the plugin under the same name and a different version.

![image](https://f.cloud.github.com/assets/3426622/470724/8e49edd4-b701-11e2-924d-fb220e333d3a.png)

The download URL of the "new" Version still somehow points to the initial version - so even if you select a more recent version in NINJA-IDE plugin manager the old version gets pulled.

Expected Result: Single Instance of the Plugin - old version gets overwritten by most recent Version

**Comment by pmourelle** [comment]
Hi, thanks for pointing this out!
It's obviously a serious bug (it's truly a bad implementation, but keep this just between you and me and the zillion of people who can read this).
A refactor must be done to fix it. Hopefully by the weekend we'll get our hands dirty and finish this business mafia-style (¿?).
The expected result, as you say, is that just 1 version should be available for everyone... else but you. You should be able to see your previous versions in case you screw up number version, description, or whatever.

Summing up: more news on weekend. thanks.

**Comment by blackcoffeerider** [comment]
Hi Pedro,
i checked out the git repo of the website when I found this - to see if I can quickly spot the error but didn't get anywhere within the 20 minutes to patch anything usefull... I'm just getting started with Django so I'm no real help sorry - looks like there is no check at the Plugin model if something under the same ID has already been created so I guess it is at least missing some Django META stuff in there....
